runtime/include/chpl-mem-array.h

```diff
@@ -96,12 +96,18 @@ void* chpl_mem_array_alloc(size_t nmemb, size_t eltSize,
   chpl_memhook_malloc_post(p, nmemb, eltSize, CHPL_RT_MD_ARRAY_ELEMENTS,
                            lineno, filename);
 #ifdef HAS_GPU_LOCALE
+  // The CUDA driver uses DMA to transfer page-locked memory to the GPU; if
+  // memory is not page-locked it must first be transfered into a page-locked
+  // buffer, which degrades performance. So in the array_on_device mode we
+  // choose to page-lock such memory even if it's on the host-side.
+  #ifdef CHPL_GPU_MEM_STRATEGY_ARRAY_ON_DEVICE
+  cudaHostRegister(p, size, cudaHostRegisterPortable);
```

Comment:
You may already be planning to do that, but this should be a `chpl_gpu_mem_register` or something in the runtime GPU interface that calls `chpl_gpu_impl_mem_register` in different backend implementation(s).

Reply:
Good idea, I've updated to do that in this commit: https://github.com/chapel-lang/chapel/pull/21483/commits/2ba18d178b8250864b5dc15f11fc898e1a6489e0